apix_documentation_swagger_v4.yaml

```diff
@@ -0,0 +1,1186 @@
+        This endpoint gets the last migration attempt of a bill.
+      operationId: getMigration
+      parameters:
+        - name: id
+          in: path
+          description: ID of the bill whose migration information you wish to
+            obtain
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: successful response
+          content:
+            application/vnd.regalii.v4.0+json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    allOf:
+                      - $ref: '#/components/schemas/Migration'
+
+                example:
+                  data:
+                    id: b4aa7d78-6f16-4ddd-887d-9f5c0b730a26
+                    type: migration
+                    attributes:
+                      status: failed
+                      requested_at: '2018-06-15 14:16:16 UTC'
+                      migrated_at: '2018-06-15 16:34:54 UTC'
+                      error:
+                        code: InvalidCredentials
+                        detail: Invalid login or password
+
+  /migrations/{id}/challenges:
+    get:
+      tags:
+        - Challenges
+      summary: Gets all challenges associated with a specific migration
+      description: |
+        If the migration requires the end user to answer some challenges, the
+        customer can request the challenges associated to one specific
+        migration by using this endpoint.
+      operationId: getMigrationsChallenges
+      parameters:
+        - name: id
+          in: path
+          description: ID of the migration whose challenges you wish to obtain
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: successful response
+          content:
+            application/vnd.regalii.v4.0+json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    type: array
+                    items:
+                      anyOf:
+                        - $ref: '#/components/schemas/MFA_Challenge_Question'
+                        - $ref: '#/components/schemas/MFA_Challenge_Code'
+                        - $ref: '#/components/schemas/MFA_Challenge_Image'
+                        - $ref: '#/components/schemas/MFA_Challenge_Image_Choices'
+                        - $ref: '#/components/schemas/MFA_Challenge_Choices'
+                allOf:
+                  - $ref: '#/components/schemas/MetaData'
+                  - $ref: '#/components/schemas/NavigationLinks'
+
+                example:
+                  data:
+                    - id: 3a798711-d352-4ee5-b35d-383fd23e7edc
+                      type: challenge
+                      attributes:
+                        kind: code
+                        created_at: '2018-06-15 14:16:16 UTC'
+                        expires_at: null
```

Comment:
The code expires in 3 minutes, I don't think this should be null

Expiration times [here](https://github.com/regalii/api-x/blob/develop/app/models/challenge.rb#L99)